[BUG] org.opensearch.snapshots.SearchableSnapshotIT.testPruneFileCacheOnIndexDeletion is flaky

**Describe the bug**
Test failure - 
```
Error Message
java.lang.AssertionError: timed out waiting for green state
Stacktrace
java.lang.AssertionError: timed out waiting for green state
	at __randomizedtesting.SeedInfo.seed([1ED65DC09068BADB:C6BEB40884E3C77A]:0)
	at org.junit.Assert.fail(Assert.java:89)
	at org.opensearch.test.OpenSearchIntegTestCase.ensureColor(OpenSearchIntegTestCase.java:1001)
	at org.opensearch.test.OpenSearchIntegTestCase.ensureGreen(OpenSearchIntegTestCase.java:932)
	at org.opensearch.test.OpenSearchIntegTestCase.ensureGreen(OpenSearchIntegTestCase.java:921)
	at org.opensearch.snapshots.SearchableSnapshotIT.createIndexWithDocsAndEnsureGreen(SearchableSnapshotIT.java:279)
	at org.opensearch.snapshots.SearchableSnapshotIT.testPruneFileCacheOnIndexDeletion(SearchableSnapshotIT.java:534)
```

**To Reproduce**
https://build.ci.opensearch.org/job/gradle-check/12588/

**Expected behavior**
Test should pass.

--- a/server/src/internalClusterTest/java/org/opensearch/snapshots/SearchableSnapshotIT.java
+++ b/server/src/internalClusterTest/java/org/opensearch/snapshots/SearchableSnapshotIT.java
@@ -522,7 +522,6 @@ public final class SearchableSnapshotIT extends AbstractSnapshotIntegTestCase {
         return stats.getUsed().getBytes() == 0L && stats.getActive().getBytes() == 0L;
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/opensearch-project/OpenSearch/issues/6738")
     public void testPruneFileCacheOnIndexDeletion() throws Exception {
         final String snapshotName = "test-snap";
         final String repoName = "test-repo";
@@ -531,7 +530,7 @@ public final class SearchableSnapshotIT extends AbstractSnapshotIntegTestCase {
         final Client client = client();
         final int numNodes = 2;
 
-        internalCluster().ensureAtLeastNumSearchNodes(numNodes);
+        internalCluster().ensureAtLeastNumSearchAndDataNodes(numNodes);
         createIndexWithDocsAndEnsureGreen(1, 100, indexName1);
 
         createRepositoryWithSettings(null, repoName);

